File: 12/HighlyDivisibleTriangularNumber.cpp
#include<bits/stdc++.h>
using namespace std;
unsigned long long Tri(int n){ return (n*(n+1))>>1;}
int main()
{
    int Starter = 200;
    unsigned long long Value  = 0;
    int NOD = 500; //Number of divisiors required
    for(;;)
    {
        unsigned long long Current = Tri(Starter);
        int Count = 0;
        for(unsigned long long p=1; p<=Current; ++p)
        {
            if(Current%p == 0) Count++;
            if(Count>NOD){
                cout<<Starter<<"th triangle number "<<Current<<endl;
                return 0;
            }
        }
        Starter++;
    }
}

File: 12/HighlyDivisibleTriangularNumber.py
#!/usr/bin/python3

def Tri(n):
    p = int(n*(n+1)/2.);
    return p
	
def main():
    Starter = 20;
    Value  = 0;
    NOD = 200; #//Number of divisiors required
    while(True):
        Current = Tri(Starter)
        #print('Current tri is ',Current)
        Count = 0
        for p in range(1,Current+1):
            if Current % p == 0:
                Count += 1
                if(Count>NOD):
                    print(Starter, "th triangle number ",Current)
                    return 0;

        Starter += 1;

y = main()
